### ISSUE TITLE
Zip Download for Single Folder

### ISSUE THREAD

**Issue by julmot** [jsdelivr/jsdelivr#18040]
Since many packages have a `dist` folder in their repository, and many users are still creating websites without npm or Bower, it would be helpful to offer a ZIP download of single folders. I know that there's already the option to download a `.tgz` file. But it contains the whole repository and also needs to be extracted twice on Windows.

**Comment by MartinKolarik** [comment]
>needs to be extracted twice on Windows.

That depends on (and is more of a problem with) your software. There isn't a good reason why it couldn't be extracted in one step.

The problem with offering ZIP packages for individual directories is that we would need to generate those ourselves, which involves a fair amount of work. Currently we simply link to the existing packages on npm.

**Comment by julmot** [comment]
I guess this library would remove a bit of complexity: https://stuk.github.io/jszip/

**Comment by MartinKolarik** [comment]
Are you suggesting downloading all files separately and generating the ZIP in the browser?

**Comment by julmot** [comment]
Whatever is the easiest to implement. I guess saving and caching it would cause an overhead that isn't justifiable in comparison to just waiting a short time for a ZIP you probably only need once. So I'd probably suggest writing a simple server-side script that uses JSZip to extract the existing `.tgz` file (or directoy use the unpackaged code if this is easier), and package a specific folder passed (commonly `dist`) passed along as query parameter and directly return that output.

**Comment by MartinKolarik** [comment]
I'm note sure how well would the client-side script work, but I can tell you doing this on server-side would likely cause worse UX than now.

The full packages are already prepared and cached on the CDN, so downloading starts instantly. If we wanted to generate a custom ZIP package on demand, we would have to:
1. Fetch the full package from npm 
2. Extract it
3. Pack the requested directory
4. Serve it to the user (maybe streaming would work here, so we could be packing and serving at the same time) - and this would be serving from our server, not from CDN

In other words, it would add delay equal to the time it takes to do steps 2 - 4.

**Comment by MartinKolarik** [comment]
@julmot I'm going to close this because a server-side solution would require too much work for small/no UX improvement. If you would like to try implementing the client-side version, feel free to check our website code at https://github.com/jsdelivr/www.jsdelivr.com
